src/extension.ts

```diff
@@ -1,49 +1,52 @@
 'use strict';
-// The module 'vscode' contains the VS Code extensibility API
-// Import the module and reference it with the alias vscode in your code below
-import { workspace, commands, ExtensionContext, window } from 'vscode';
+
 import { dirname } from 'path';
+import { commands, ExtensionContext, window, workspace } from 'vscode';
 
-// this method is called when your extension is activated
-// your extension is activated the very first time the command is executed
 export function activate(context: ExtensionContext) {
-  // The command has been defined in the package.json file
-  // Now provide the implementation of the command with  registerCommand
-  // The commandId parameter must match the command field in package.json
   let disposable = commands.registerCommand('extension.ignoreGitignore', async () => {
-    // The code you place here will be executed every time your command is executed
 
-    // Just use the root one for now
-    const gitignores = await workspace.findFiles('**/.gitignore')
+    // Get a list of all .gitignore files in the project
+    const gitignores = await workspace.findFiles('**/.gitignore');
 
-    let path;
-    if (!gitignores.length) {
+    if (!gitignores) {
       window.showInformationMessage('No .gitignore files found in current workspace.');
       return;
     }
 
-    const excludeObj: { [glob: string]: boolean } = {};
     const gitignoreDocumentPromises = gitignores.map((file) => {
       return workspace.openTextDocument(file.path);
     });
 
     const gitignoreDocuments = await Promise.all(gitignoreDocumentPromises);
 
+    const excludeObj: { [glob: string]: boolean } = {};
+
     gitignoreDocuments.forEach((doc) => {
       const relativePath = workspace.asRelativePath(doc.fileName);
       const dir = dirname(relativePath);
 
       // Let's do search for now
       for (let lineIndex = 0; lineIndex < doc.lineCount; lineIndex++) {
-        const lineText = doc.lineAt(lineIndex).text;
+        let lineText = doc.lineAt(lineIndex).text;
 
-        if (lineText.length) {
-          let lineTextWithPath = lineText;
+        if (lineText.length && !lineText.startsWith('#')) {
+          const negated = lineText.startsWith('!');
+          if (negated) {
+            lineText = lineText.substr(1);
+          }
+
+          // pattern starts at project root
+          let glob = lineText.startsWith('/')
+            ? lineText.substr(1)
+            : `**/${lineText}`;
```

Comment:
We also don't want to prefix `**/` if `lineText` contains any non-trailing `/`:

>• If the pattern ends with a slash, it is removed for the purpose of the following description, ...
>• If the pattern does not contain a slash /, ...
>• Otherwise, Git treats the pattern as a shell glob suitable for consumption by fnmatch(3) with the FNM_PATHNAME flag: wildcards in the pattern will not match a / in the pathname. For example, "Documentation/*.html" matches "Documentation/git.html" but not "Documentation/ppc/ppc.html" **or "tools/perf/Documentation/perf.html"**.